addons/web/static/src/views/form/form_compiler.js

```diff
@@ -224,8 +224,7 @@ export class FormCompiler extends ViewCompiler {
         const displayClasses = sheetNode
             ? `d-flex {{ uiService.size < ${SIZES.XXL} ? "flex-column" : "flex-nowrap h-100" }}`
             : "d-block";
-        const stateClasses =
-            "{{ props.record.isVirtual or props.record.isDirty ? 'o_form_dirty' : 'o_form_saved' }}";
+        const stateClasses = "{{ props.record.isDirty ? 'o_form_dirty' : 'o_form_saved' }}";
```

Comment:
need to check with @mcm-odoo 

it's weird to have a brand new record with "o_form_saved" classname.

Also this needs a test.

Note: maybe the culprit here is the tour helper, not the form